twilio/base/client_base.py

```diff
@@ -132,9 +146,19 @@ async def request_async(
                 "http_client must be asynchronous to support async API requests"
             )
 
-        auth = self.get_auth(auth)
+
         headers = self.get_headers(method, headers)
-        uri = self.get_hostname(uri)
+
+        ##If credential provider is provided by user, get the associated auth strategy
+        ##Using the auth strategy, fetch the auth string and set it to authorization header
+        auth_strategy = None ##Initialization
```

Comment:
We should have this similar to [above](https://github.com/twilio/twilio-python/pull/815/files#diff-4a42d5c4695b732baef521f6b7ce02fffa66b18efabb3ca7527ae184c210d823R96-R101)

Reply:
updated